app/models/spree/product_decorator.rb

```diff
@@ -165,6 +165,15 @@ def variants_distributed_by(order_cycle, distributor)
     order_cycle.variants_distributed_by(distributor).where(product_id: self)
   end
 
+  def import_date
+    # Get the most recent import_date of a product's variants
+    imports = []
+    variants.each do |v|
+      imports.append(v) unless v.import_date.blank?
+    end
+    imports.sort_by(&:import_date).last.try(:import_date)
+  end
```

Comment:
You can use `#map()` here instead of `#each()`, so you can avoid variable assignations and array appends:
```ruby
def import_date
  variants.map do |variant|
    next if variant.import_date.blank?
    variant.import_date
  end.sort.last
end
```
You gain in readability too. Didn't try the code but you get the gist of it.